docs/topics/gradle/gradle-compiler-options.md

```diff
@@ -136,6 +136,340 @@ tasks.named('compileKotlin', org.jetbrains.kotlin.gradle.tasks.KotlinCompilation
 </tab>
 </tabs>
 
+### Migrate from `kotlinOptions{}` to `compilerOptions{}` {collapsible="true"}
+
+Before Kotlin 2.2.0, you could configure compiler options using the `kotlinOptions{}` block. Since the `kotlinOptions{}`
+block is deprecated in Kotlin 2.2.0, this section provides guidance and recommendations for migrating your build
+scripts to use the `compilerOptions{}` block instead:
+
+* [Centralize compiler options and use types](#centralize-compiler-options-and-use-types)
+* [Migrate away from `android.kotlinOptions`](#migrate-away-from-android-kotlinoptions)
+* [Migrate `freeCompilerArgs`](#migrate-freecompilerargs)
+
+#### Centralize compiler options and use types
+
+Whenever possible, configure compiler options at the [extension level](#extension-level), and override them for specific tasks
+at [compilation unit level](#compilation-unit-level).
+
+To make your configuration safer and more expressive, avoid using raw strings and prefer strongly typed values. For example, if you have:
+
+<tabs group="build-script">
+<tab title="Kotlin" group-key="kotlin">
+
+```kotlin
+plugins {
+    kotlin("jvm") version "2.2.0"
+}
+
+tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {
+    kotlinOptions {
+        jvmTarget = "17"
+        languageVersion = "2.2"
+        apiVersion = "2.2"
+    }
+}
+```
+
+</tab>
+<tab title="Groovy" group-key="groovy">
+
+```kotlin
+plugins {
+    id 'org.jetbrains.kotlin.jvm' version '2.2.0'
+}
+
+tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).configureEach {
+    kotlinOptions {
+        jvmTarget = '17'
+        languageVersion = '2.2'
+        apiVersion = '2.2'
+    }
+}
+```
+
+</tab>
+</tabs>
+
+After migration, this becomes:
+
+<tabs group="build-script">
+<tab title="Kotlin" group-key="kotlin">
+
+```kotlin
+plugins {
+    kotlin("jvm") version "2.2.0"
```

Comment:
BTW do we update this version automatically on each new Kotlin release?

Reply:
Added variables!